### ISSUE TITLE
tmux gets unrecoverably stuck and use 100% CPU

### ISSUE THREAD

**Issue by zozowell** [tmux/tmux#2302]
### Issue description

tmux entered a freeze state time to time, unrecoverable. It uses upto 100% CPU at that state. I had to kill tmux process to recover. Following is backtrace of stuck tmux process:

```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00000001089f2e80 tmux`grid_get_cell1
    frame #1: 0x00000001089f5617 tmux`grid_line_length + 119
    frame #2: 0x0000000108a2ae50 tmux`window_copy_cursor_previous_word_pos + 336
    frame #3: 0x0000000108a2ccf7 tmux`window_copy_synchronize_cursor_end + 135
    frame #4: 0x0000000108a2c9ad tmux`window_copy_set_selection + 93
    frame #5: 0x0000000108a2b988 tmux`window_copy_scroll_down + 168
    frame #6: 0x0000000108a2b3e4 tmux`window_copy_cursor_up + 132
    frame #7: 0x0000000108a2f30d tmux`window_copy_cmd_scroll_up + 45
    frame #8: 0x0000000108a29ea7 tmux`window_copy_command + 327
    frame #9: 0x00000001089dfc6e tmux`cmd_send_keys_exec + 350
    frame #10: 0x00000001089dd168 tmux`cmdq_next + 920
    frame #11: 0x0000000108a146cd tmux`server_loop + 61
    frame #12: 0x0000000108a07048 tmux`proc_loop + 56
    frame #13: 0x0000000108a13fb9 tmux`server_start + 553
    frame #14: 0x00000001089cd5f2 tmux`client_main + 1058
    frame #15: 0x0000000108a1d2d1 tmux`main + 1665
    frame #16: 0x00007fff73636cc9 libdyld.dylib`start + 1
    frame #17: 0x00007fff73636cc9 libdyld.dylib`start + 1
```

The trigger operation is unknown, but have following clue:
- It happens after long time using a tmux session, with multiple windows.
- Most likely happens in copy mode.
- It is triggered by some tmux commands for sure, never saw it happened when I worked in a terminal occupying app, like emacs.

This issue is likely reported before in issue: #2253 , but didn't get an conclusion there. I love tmux very much, but this thing happens very often now, make me very nervous and frustrated, especially when I'm heavily working with lots sessions / remote connections / unsaved notes etc...  Much appreciate if anyone can look into it, thanks.

### Required information

Please provide the following information:

* tmux version (`tmux -V`). : tmux 3.1b
* Platform (`uname -sp`). : Darwin i386
* $TERM inside and outside of tmux (`echo $TERM`). : xterm-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`). 
[tmux-client-23754.log](https://github.com/tmux/tmux/files/4906255/tmux-client-23754.log)
[tmux-out-23756.log](https://github.com/tmux/tmux/files/4906256/tmux-out-23756.log)
[tmux-server-23756.log](https://github.com/tmux/tmux/files/4906257/tmux-server-23756.log)

**Comment by nicm** [comment]
Does this still happen with master?

**Comment by zozowell** [comment]
thanks @nicm ! I never tried master, will do now, need some time before getting back.

**Comment by JimFawkes** [comment]
I also experienced this behavior a few times recently. In all cases I had several sessions running for a long time. Unfortunately I currently have no additional information, but will add details if it happens again. My system info is the same as listed above.

**Comment by nicm** [comment]
Make sure you have afe4ea4250073e482c6ec6accfc539f873df6977.

**Comment by yujinyuz** [comment]
Will this be fixed in 3.2?

**Comment by zozowell** [comment]
An update. It never happened again in latest master since I first reported it in this thread.
